vcl/main.vcl

```diff
@@ -135,6 +135,13 @@ sub vcl_fetch {
         return (pass);
     }
 
+    # If we've gotten an error after the restarts on an ESI page, then we'll
+    # replace the error contents with a blank page so that we don't render the
+    # error contents into the final document.
+    if (req.url ~ "^/_esi/" && (beresp.status == 500 || beresp.status == 503)) {
```

Comment:
I'd personally check the status code first, since it's a more path to be a non-50x error and it's faster to compare an int vs a regular expression. :)

Alternatively, you could check against `beresp.http.Warehouse-ESI-Enable`, no? And do this for all ESI? Or is this intended for very specific endpoints?